src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1362,16 +1356,7 @@ impl Store {
                   .await
               }
               directory::Entry::Directory(d) => {
-                // NB: This overcounts the size of the digest, since it includes the size of directory digests.
-                if can_be_immutable
-                  && d
-                    .tree()
-                    .digests()
-                    .iter()
-                    .map(|d| d.size_bytes)
-                    .sum::<usize>()
-                    > IMMUTABLE_DIR_SIZE_LIMIT
-                {
+                if can_be_immutable && d.tree().filecount_exceeds(IMMUTABLE_DIR_FILE_COUNT_LIMIT) {
```

Comment:
This isn't `O(N)` anymore, but it's still `O(N*100)`... I'd still feel better with just memoizing it.

Reply:
Ah so memoize in `materialie_` okie dokie.

Probably worth the helper now...